gui/src-tauri/src/main.rs

```diff
@@ -159,19 +200,29 @@ fn main() -> Result<()> {
 			.then(|| jre.into_os_string())
 			.or_else(|| valid_java_paths().first().map(|x| x.0.to_owned()));
 		let Some(java_bin) = java_bin else {
-			show_error(&format!("Couldn't find a compatible Java version, please download Java {} or higher", MINIMUM_JAVA_VERSION));
-			return Ok(());
+			show_error(&format!(
+                "Couldn't find a compatible Java version, please download Java {} or higher",
+                MINIMUM_JAVA_VERSION
+            ));
+			return Ok(None);
 		};
 
 		log::info!("Using Java binary: {:?}", java_bin);
-		Some((java_bin, p))
+		return Ok(Some((java_bin, p)));
```

Comment:
is this return needed?

Reply:
It is:
```
mismatched types
expected enum `Result<std::option::Option<_>, ErrReport>`
   found enum `std::option::Option<_>`
```

Reply:
Ah, no, if I remove the return, we're good XD

Reply:
That's the type, not the `return ` part

Reply:
Yeah I realized that lol